Feature/UI Deployment and Request Structure

EnricoBausenhart opened the pull request:
## Type of Change

<!-- Mark the appropriate option with an "x" -->

- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [x] ♻️ Code refactor
- [ ] 📝 Documentation update
- [ ] 🚀 Performance improvement
- [ ] ✅ Test addition/update
- [ ] 🧹 Chore/maintenance
- [ ] 📐 API Spec update
- [ ] Other (please describe):

## Changes

- AWS deployment: 
   - nip.io reached a limit of requested certificates for the provided url -> sslip.io is now used to host the deployment
   - use existing elastic ip "meetatmensa-ip" in the AWS Lab to tackle the problem with dynamic URL's (was a blocker for Auth0 allowed callbacks and CORS problems)
   - use a better ec2 instance due to performance issues with t2.micro
   - add database images to compose.aws.yml to complete setup
   - add AWS Url to allowed origins of the Gateway

- Client Dockerfile
   - as we have different setups we need different API_BASE_URL's within our vite application -> normally vite bakes these env variables into the application at build time
   - use of a small shell script entrypoint.sh to inject the variable dynamically
   - environment variable added in compose.yml for local deployment
   - environment variable renamed in compose.aws.yml
   - environment variable added in values.yaml (helm) of the client
   - Result: one Dockerfile can be used for all deployments

- UI Refactoring
   - remove unused subpaths like /preferences
   - add userID to create match request 
   - added team logo to the drawer
   - display Skeletons while requests are loading for better user experience
   - mock data can now be enabled with a VITE env variable for testing
   - improved handling of initial login and handling of userID within the application
   - refactoring of request code architecture -> separate service for each of the types (user / matchrequests / matches)
   - several small bug fixes
   
 - Networking fixes
   - Moved deployment back to Chair's Rancher after temporarily moving it to an instance on a personal server during the outage.
   - Configured ingress, TLS, and DNS on the Chair's Rancher.

EnricoBausenhart committed:
refactor: move mock data to separate files

EnricoBausenhart committed:
feat: add team logo to drawer

EnricoBausenhart committed:
feat: add userID context

EnricoBausenhart committed:
refactor: optimize code regarding api requests
also add skeletons for better user experience

EnricoBausenhart committed:
Merge branch 'main' of github.com:AET-DevOps25/team-devoops into refactor/ui-cleanup

EnricoBausenhart committed:
feat: add gateway port to dockerfile

stark-cybertech committed:
fix: add temporary support for CORS

EnricoBausenhart committed:
feat: disable mock data in deployment

stark-cybertech committed:
Merge branch 'main' into refactor/ui-cleanup

stark-cybertech committed:
fix: add meetatmensa.com to CORS

stark-cybertech committed:
feat: set address for gateway

stark-cybertech committed:
fix: add http

stark-cybertech committed:
fix: correct public address

stark-cybertech committed:
fix: correct CORS entry

stark-cybertech committed:
feat: enable deployment to starnet rancher

EnricoBausenhart committed:
feat: adapt subpage paths

EnricoBausenhart committed:
feat: improve requests to the gateway including mock data switch

EnricoBausenhart committed:
feat: optimize code structure and fix minor issues

EnricoBausenhart committed:
fix: typescript errors in docker build

EnricoBausenhart committed:
feat: dynamically inject api base url

EnricoBausenhart committed:
feat: add config template for dynamically injected url

EnricoBausenhart committed:
feat: adapt base url in docker compose file

EnricoBausenhart committed:
feat: use existing elastic ip in aws deployment to fix issue with dynamic ips

EnricoBausenhart committed:
feat: use dynmaic api base url in helm deployment

EnricoBausenhart committed:
feat: add aws url to allowed origins of gateway

EnricoBausenhart committed:
fix: elastic ip allocation

EnricoBausenhart committed:
fix: remove duplicated api version path

EnricoBausenhart committed:
feat: add databases to aws setup and test new valu einjection

EnricoBausenhart committed:
fix: test injection

EnricoBausenhart committed:
fix: test with different image

EnricoBausenhart committed:
fix: move back to latest client image

EnricoBausenhart committed:
feat: adapt container names

EnricoBausenhart committed:
feat: add database images

EnricoBausenhart committed:
feat: set ansibl ehost

EnricoBausenhart committed:
fix: name of database images

EnricoBausenhart committed:
feat: test bigger ec2 instance

EnricoBausenhart committed:
feat: use smaller sized ec2 instace for cost reasons

EnricoBausenhart committed:
feat: update compose file due to certificate limits in nip io

EnricoBausenhart committed:
fix: revert to t3.small

EnricoBausenhart committed:
feat: add new allowed origin to gateway

EnricoBausenhart committed:
Feature/aws setup (#84)

stark-cybertech commented:
> @stark-cybertech Do we have go back to the "latest" tag in charts/client/values.yaml? The feature tag was only for testing, right?

Yes! Please go back to latest!

stark-cybertech reviewed (changes requested):
This looks excellent, but will not deploy successfully to the chair's Rancher instance yet because of the changes made to networking.

I'm working on shifting the DNS records and setting up a reverse proxy on Chair rancher, lets wait for that before merging this.

EnricoBausenhart committed:
feat: update readme and adapt env example file

EnricoBausenhart committed:
Merge branch 'refactor/ui-cleanup' of github.com:AET-DevOps25/team-devoops into refactor/ui-cleanup

EnricoBausenhart committed:
docs: add local docker development instructions to the client readme

stark-cybertech committed:
pain: set CORS origin to * according to "best practices"

This is a bad idea, but the chair insists we deploy to their hardware.... https://imgur.com/a/OkhqY5g

stark-cybertech committed:
feat: prototype networking for deployment

stark-cybertech committed:
fix: deploy SHA tag instead of latest

This fixes an issue where helm upgrade fails to actually fetch the latest image when the tag hasn't changed

stark-cybertech committed:
fix: propagate tags to subcharts

stark-cybertech committed:
Merge branch 'refactor/ui-cleanup' into feature/networking

EnricoBausenhart committed:
fix: move image tag back to latest

stark-cybertech committed:
Merge branch 'refactor/ui-cleanup' into feature/networking

EnricoBausenhart committed:
Feature/networking (#86)

stark-cybertech committed:
fix: do not overwrite database images

stark-cybertech reviewed (changes requested):
Just tested the CI/CD action and deployment to rancher is working smoothly!

Still struggling with the following issue though:

### Where:
GET @ api/v2/user/me
### When:
On start when first fetching user-id
### Error:
Browser reports "CORS request did not succeed"
### Cause:
Special character "|" in URL, part of the auth0 ID.
### Solution:
Encode "|" as "%7C"

EnricoBausenhart committed:
feat: esnure auth encoding and waiting for users/me request to finish

stark-cybertech reviewed (approved):
Deployment works and no major issues with the client! Ready to deploy! 🤘🤘
